### ISSUE TITLE
problem download assets at install

### ISSUE THREAD

**Issue by StefPiturca** [safing/portmaster#1186]
Hello i got a problem with the setup it always fails to download the portmaster 3000 mb assets and i can't at all install the app or use it because of this problem.Any way to solve the issue?

**Comment by dhaavi** [comment]
Hey @StefPiturca, our update server is quite capable, so not sure why this happens?

Where are you located? (Country / ISP)

**Comment by StefPiturca** [comment]
> Hey @StefPiturca, our update server is quite capable, so not sure why this happens?
> 
> Where are you located? (Country / ISP)

I'm from Romania,and i live in Craiova.

**Comment by StefPiturca** [comment]
is it a country problem?

**Comment by h6fYe34KFSiH** [comment]
Is it possible to release a complete installation package?

**Comment by Raphty** [comment]
we are thinking about an offline installer - it would have limited information on first start, and will need an update - but you could use it and the update wont be as big

**Comment by Jarfeh** [comment]
I believe this might help with explaining the issue. I'm in the US and experience this issue as well. 
I manually downloaded portmaster-start.exe from the link provided in [the other issue](https://github.com/safing/portmaster/issues/1181#issuecomment-1515833707) and ran:
`portmaster-start.exe --data C:\ProgramData\Safing\Portmaster update`

The following was printed after running:
`C:\ProgramData\Safing\Portmaster>[pmstart] 2023/05/18 11:12:55 WARNING: error loading indexes: failed to fetch index all/intel/intel.json: failed to make request to "https://updates.safing.io/all/intel/intel.json": Get "https://updates.safing.io/all/intel/intel.json": tls: failed to verify certificate: x509: certificate has expired or is not yet valid:
Error: failed to fetch index all/intel/intel.json: failed to make request to "https://updates.safing.io/all/intel/intel.json": Get "https://updates.safing.io/all/intel/intel.json": tls: failed to verify certificate: x509: certificate has expired or is not yet valid:`

Visiting the URL successfully retrieves the json, whatever its purpose is for, but appears to fail for portmaster-start.exe

**Comment by iulko** [comment]
> we are thinking about an offline installer - it would have limited information on first start, and will need an update - but you could use it and the update wont be as big

Which part of the program would need that update to work? SPN?

**Comment by Raphty** [comment]
The Portmaster installer is not updated for each Portmaster update because the signing process of the installers takes time. (updates are signed as well but differently)

So the installer is just a core that then downloads basically all of Portmaster.

The offline installer would work in a similar way so it will get outdated and will update after launch.
It will also not have all the filter lists up to date,  which we update each hour.

**Comment by iulko** [comment]
> The Portmaster installer is not updated for each Portmaster update because the signing process of the installers takes time. (updates are signed as well but differently)
> 
> So the installer is just a core that then downloads basically all of Portmaster.
> 
> The offline installer would work in a similar way so it will get outdated and will update after launch. It will also not have all the filter lists up to date, which we update each hour.

I belive the point everybody is making about offline installer is to bundle current lists/plugins etc when you release new installer so the installer have things that are current at the time of release not the other way around

**Comment by Raphty** [comment]
This is what I said, they would probably be released together with the installers and would be out of date basically the next day.

We hear the requests, and we will look into offline install options in the future.

**Comment by Raphty** [comment]
@StefPiturca did the install work now? did you find out what the issue was?

**Comment by earmin** [comment]
Having similar issue from Canada, installation error and CMD's error atatched.
![image](https://github.com/safing/portmaster/assets/3898995/2973e98d-bc0c-493f-a922-3641bcdb70b0)

**Comment by earmin** [comment]
Thanks @dhaavi,
That's wired. I have Simplewall installed on that PC and in several tries it didn't bring up the firewall popup to give it access, like it's not noticing this request or something. But I was able to open "https://updates.safing.io/all/intel/intel.json" itself on the browser. 
Today I tried again and it worked and finally installed. Thank you! 🙂
![image](https://github.com/safing/portmaster/assets/3898995/e585ff2a-0ee7-4f92-979b-4b0e8ce719f5)
